app/scripts/models/reliers/relier.js

```diff
@@ -44,11 +52,18 @@ define([
      *         // do overriding behavior here.
      *       });
      * }
+     *
+     * @method fetch
      */
     fetch: function () {
       var self = this;
       return p()
         .then(function () {
+          // parse the resume token before importing any other data.
+          // query parameters and server provided data override
+          // resume provided data.
+          self.populateFromStringifiedResumeToken(self.getSearchParam('resume'));
```

Comment:
What happens if there's no resume token in the URL?  IIUC it will bottom out at a call to `parse` which returns undefined, yielding no fields and hence not populating any data.  But we should at least add an explicit test for this behaviour.